File: server/battle/src/main/java/info/chenliang/moba/message/Position3d.java
package info.chenliang.moba.message;
import java.io.*;
import info.chenliang.talky.*;
public class Position3d{
public void serialize(DataOutputStream dos) throws Exception{
FieldMark fm = new FieldMark(0);
dos.write(fm.getData());
}
public void deserialize(DataInputStream dis) throws Exception{
FieldMark fm = new FieldMark(0);
dis.read(fm.getData());
}
}

File: client/Assets/Scripts/Message/info/chenliang/moba/message/Position3d.cs
using System;
using System.IO;
using MiscUtil;
using MiscUtil.IO;
using MiscUtil.Conversion;
using Talky;
namespace info.chenliang.moba.message{
public class Position3d{
public void Serialize(MemoryEndianBinaryWriter writer){
FieldMark fm = new FieldMark(0);
writer.Write(fm.GetData());
}
public void Deserialize(MemoryEndianBinaryReader reader){
FieldMark fm = new FieldMark(0);
reader.Read(fm.GetData(), 0, fm.GetData().Length);
}
}
}
